packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -129,19 +129,32 @@ class _DatePickerHeader extends StatelessWidget {
     Widget yearButton = new _DateHeaderButton(
       color: backgroundColor,
       onTap: Feedback.wrapForTap(() => _handleChangeMode(DatePickerMode.year), context),
-      child: new Text(localizations.formatYear(selectedDate), style: yearStyle),
+      child: new Semantics(
+        selected: mode == DatePickerMode.year,
+        child: new Text(localizations.formatYear(selectedDate), style: yearStyle),
+      ),
     );
     Widget dayButton = new _DateHeaderButton(
       color: backgroundColor,
       onTap: Feedback.wrapForTap(() => _handleChangeMode(DatePickerMode.day), context),
-      child: new Text(localizations.formatMediumDate(selectedDate), style: dayStyle),
+      child: new Semantics(
+        selected: mode == DatePickerMode.day,
+        child: new Text(localizations.formatMediumDate(selectedDate), style: dayStyle),
+      ),
     );
 
     // Disable the button for the current mode.
-    if (mode == DatePickerMode.day)
-      dayButton = new IgnorePointer(child: dayButton);
-    else
-      yearButton = new IgnorePointer(child: yearButton);
+    if (mode == DatePickerMode.day) {
+      dayButton = new IgnorePointer(
+        child: dayButton,
+        ignoringSemantics: false,
+      );
+    } else {
+      yearButton = new IgnorePointer(
+        child: yearButton,
+        ignoringSemantics: false,
+      );
+    }
```

Comment:
this would be more idiomatic as a switch

Reply:
although, actually, why not just put the IgnorePointer in the original widget subtree? that would be more efficient since then when it changes we wouldn't be changing the depth of the child subtree.

Reply:
Moved `IgnorePointer` to the original subtree, so it's not changing depth any more. I'm not sure how I would use `Switch`, as it comes with its own custom UI.

Reply:
I meant like a switch with case statements, not the Switch widget. :-)